recipes/boost/all/conanfile.py

```diff
@@ -752,32 +757,34 @@ def create_library_config(name):
 
     def _get_toolset_version_and_exe(self):
         compiler_version = str(self.settings.compiler.version)
+        major = compiler_version.split(".")[0]
+        if "." not in compiler_version:
+            compiler_version += ".0"
         compiler = str(self.settings.compiler)
         if self._is_msvc:
-            cversion = self.settings.compiler.version
-            if Version(str(cversion)) >= "16":
+            if Version(compiler_version) >= "16":
                 _msvc_version = "14.2"
-            elif Version(str(cversion)) >= "15":
+            elif Version(compiler_version) >= "15":
                 _msvc_version = "14.1"
             else:
-                _msvc_version = "%s.0" % cversion
+                _msvc_version = compiler_version
             return "msvc", _msvc_version, ""
         elif self.settings.os == "Windows" and self.settings.compiler == "clang":
             return "clang-win", compiler_version, ""
         elif self.settings.os == "Emscripten" and self.settings.compiler == "clang":
             return "emscripten", compiler_version, self._cxx
         elif self.settings.compiler == "gcc" and tools.is_apple_os(self.settings.os):
             return "darwin", compiler_version, self._cxx
-        elif compiler == "gcc" and compiler_version[0] >= "5":
```

Comment:
NOTE: previous code had a bug! if we have GCC >= 10, it would compare "1" >= "5". also, it did string comparison instead of integer.